### PR TITLE
restart pods with injected certs when the certs rotate 

### PR DESCRIPTION
**Pull request by staebler**
The hiveadmission and hiveapi pods use certs that may change while the pods are running. When the certs change, the pods will no longer be able to do their work. The pods need to be restarted to pick up the new certs.

### CODE REVIEW

**Comment by staebler** [comment]
https://issues.redhat.com/browse/CO-782

File: pkg/certs/certs.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		if err != nil {
+			log.WithError(err).WithField("filename", fullFilename).Warning("Unable to read initial content of cert file")
+			continue // intentionally ignore errors
+		}
+		files[fullFilename] = fileBytes
+	}
+	obs.AddReactor(func(filename string, action fileobserver.ActionType) error {
+		log.WithField("filename", filename).Info("exiting because cert changed")
+		// TODO: Exit more cleanly, allowing for cleanup by closing contexts.
+		os.Exit(1)
```
Comment (joelddiaz) [inline pkg/certs/certs.go]: Why exit non-zero? I wouldn't expect the cert change to be considered an error condition.

Comment (staebler) [reply pkg/certs/certs.go]: Makes sense. I will change it.

File: pkg/certs/certs.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		return nil
+	}, files, fullFilenames...)
+
+	stop := make(chan struct{})
+	go func() {
+		log.WithField("files", fullFilenames).Info("running file observer")
+		obs.Run(stop)
+		log.Error("file observer stopped")
```
[the remark this answers is not in the document]
Comment (staebler) [reply pkg/certs/certs.go]: Yeah, that seems like a good idea.

**Comment by staebler** [comment]
> this all looks good to me. can we squash the last two commits?

I can squash the last commit. I prefer to keep the vendor commit separate in order to isolate the noise from vendor changes from the signal of Hive code changes.

**Comment by dgoodwin** [comment]
Awesome, glad to see this one fixed.

**Pull request merged**